sycl/include/sycl/ext/intel/experimental/esimd/math.hpp

```diff
@@ -39,8 +39,8 @@ __ESIMD_API std::enable_if_t<std::is_integral<T0>::value &&
                              __ESIMD_NS::simd<T0, SZ>>
 shl(__ESIMD_NS::simd<T1, SZ> src0, U src1, Sat sat = {}) {
   using ComputationTy = __ESIMD_DNS::computation_type_t<decltype(src0), U>;
-  typename __ESIMD_DNS::simd_type<ComputationTy>::type Src0 = src0;
-  typename __ESIMD_DNS::simd_type<ComputationTy>::type Src1 = src1;
+  __ESIMD_NS::simd<T1, SZ> Src0 = src0;
```

Comment:
could you please explain a bit more for better understanding what the problem was and how this change fixes it? With this change promotion does not happen on Src0

Reply:
Yes, please. The ESIMD approach is to perform conversion and promotions on vector operands the [same way](https://en.cppreference.com/w/cpp/language/operator_arithmetic) C++ performs these on scalar operands of the same operation.